service-workers/service-worker/resources/no-dynamic-import.js

```diff
@@ -0,0 +1,19 @@
+/** @type {[name: string, url: string][]} */
+const importUrlTests = [
+  ["Module URL", "./basic-module.js"],
+  // In no-dynamic-import-in-module.any.js, this module is also statically imported
+  ["Another module URL", "./basic-module-2.js"],
+  [
+    "Module data: URL",
+    "data:text/javascript;charset=utf-8," +
+      encodeURIComponent(`export default 'hello!';`),
+  ],
+];
+
+for (const [name, url] of importUrlTests) {
+  promise_test(
+    async (t) =>
+      promise_rejects_js(t, TypeError, import(url), "Import must reject"),
```

Comment:
I think you either need await here or remove the async wrapper and return this. Otherwise the test will pass and the assert will happen too late, no?

Reply:
The async is redundant, but it should work fine. It's an arrow function without `{`, so `promise_rejects_js` is returned. I'll remove the async…